frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/rbac.tsx

```diff
@@ -0,0 +1,155 @@
+};
+
+/**
+ * Memoizes the result so it is possible to only make the request once for each access review.
+ * This does mean that the user will have to refresh the page to see updates.
+ * Function takes in the destructured resource attributes so that the cache keys are stable.
+ * `JSON.stringify` is not guaranteed to give the same result for equivalent objects.
+ * Impersonate headers are added automatically by `k8sCreate`.
+ * @param group resource group.
+ * @param resource resource string.
+ * @param subresource subresource string.
+ * @param verb K8s verb.
+ * @param namespace namespace.
+ * @param impersonateKey parameter to include in the cache key even though it's not used in the function body.
+ * @return Memoized result of the access review.
+ */
+const checkAccessInternal = _.memoize(
+  (
+    group: string,
+    resource: string,
+    subresource: string,
+    verb: K8sVerb,
+    name: string,
+    namespace: string,
+    // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+    impersonateKey: string,
+  ): Promise<SelfSubjectAccessReviewKind> => {
+    // Projects are a special case. `namespace` must be set to the project name
+    // even though it's a cluster-scoped resource.
+    const reviewNamespace =
+      group === ProjectModel.apiGroup && resource === ProjectModel.plural ? name : namespace;
+    const ssar: SelfSubjectAccessReviewKind = {
+      apiVersion: 'authorization.k8s.io/v1',
+      kind: 'SelfSubjectAccessReview',
+      spec: {
+        resourceAttributes: {
+          group,
+          resource,
+          subresource,
+          verb,
+          name,
+          namespace: reviewNamespace,
+        },
+      },
+    };
+    return k8sCreate(SelfSubjectAccessReviewModel, ssar);
+  },
+  (...args) => [...args, getActiveCluster(storeHandler.getStore().getState())].join('~'),
+);
+
+/**
+ * Provides information about user access to a given resource.
+ * @param resourceAttributes resource attributes for access review
+ * @param impersonate impersonation details
+ * @return Object with resource access information.
+ */
+export const checkAccess = (
+  resourceAttributes: AccessReviewResourceAttributes,
+  impersonate?: ImpersonateKind,
+): Promise<SelfSubjectAccessReviewKind> => {
+  // Destructure the attributes with defaults so we can create a stable cache key.
+  const {
+    group = '',
+    resource = '',
+    subresource = '',
+    verb = '' as K8sVerb,
+    name = '',
+    namespace = '',
+  } = resourceAttributes || {};
+  return checkAccessInternal(
+    group,
+    resource,
+    subresource,
+    verb,
+    name,
+    namespace,
+    getImpersonateKey(impersonate),
+  );
+};
+
+/**
+ * Hook that provides information about user access to a given resource.
+ * @param resourceAttributes resource attributes for access review
+ * @param impersonate impersonation details
+ * @return Array with isAllowed and loading values.
+ */
+export const useAccessReview = (
```

Comment:
Shall we expose `useAccessReview`, `useAccessReviewAllowed` via [api's](https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts) as well at this point or is this something we would need to decide on later? 

@andrewballantyne

Reply:
Yeah that's a good catch @invincibleJai -- per our problem and learning exercise on Thursday/Friday last week, we really should be exposing this out of the `dynamic-core-api.ts` file. Please do this change @fhlavac